header/p2p/server.go

```diff
@@ -75,69 +124,37 @@ func (serv *ExchangeServer) requestHandler(stream network.Stream) {
 
 // handleRequestByHash returns the ExtendedHeader at the given hash
 // if it exists.
-func (serv *ExchangeServer) handleRequestByHash(hash []byte, stream network.Stream) {
+func (serv *ExchangeServer) handleRequestByHash(hash []byte) ([]*header.ExtendedHeader, error) {
```

Comment:
This subroutine should return a single extended header, and it can be put into an array easily above

Reply:
True. But I wanted to avoid additional logic inside switch-case statement beside assigning. IMO, it looks clearer this way.